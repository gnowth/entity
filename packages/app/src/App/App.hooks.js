import _ from 'lodash';
import idx from 'idx';
import React from 'react';
import { DefaultProvider } from '@gnowth/default';

export default {
  useComponents(props, context) {
    return {
      DefaultProvider: props.defaults ? DefaultProvider : React.Fragment,
      IntlProvider: (props.intlProviderProps && context.intlProvider) || React.Fragment,
      ThemeProvider: (props.themeProviderProps && context.themeProvider) || React.Fragment,
    };
  },

  usePropsDefault(props, context, provider) {
    const defaults = React.useMemo(
      () => (
        _.isFunction(props.defaults)
          ? props.defaults(context.defaults)
          : { ...context.defaults, ...props.defaults }
      ),
      [context.defaults, props.defaults],
    );

    return provider === React.Fragment
      ? undefined
      : defaults;
  },

  usePropsIntl(props, context, provider) {
    const intlProviderProps = React.useMemo(
      () => (
        _.isFunction(props.intlProviderProps)
          ? props.intlProviderProps(context.intlProviderProps)
          : {
            ...context.intlProviderProps,
            ...props.intlProviderProps,
            defaultFormats: {
              ...idx(context, x => x.intlProviderProps.defaultFormats),
              ...idx(props, x => x.intlProviderProps.defaultFormats),
            },
            formats: {
              ...idx(context, x => x.intlProviderProps.formats),
              ...idx(props, x => x.intlProviderProps.formats),
            },
            messages: {
              ...idx(context, x => x.intlProviderProps.messages),
              ...idx(props, x => x.intlProviderProps.messages),
            },
          }
      ),
      [context.intlProviderProps, props.intlProviderProps],
    );

    return provider === React.Fragment
      ? undefined
      : intlProviderProps;
  },

  usePropsTheme(props, context, provider) {
    const theme = React.useMemo(
      () => (
        _.isFunction(props.themeProviderProps)
          ? props.themeProviderProps(context.themeProviderProps)
          : { ...context.themeProviderProps, ...props.themeProviderProps }
      ),
      [context.themeProviderProps, props.themeProviderProps],
    );

    return provider === React.Fragment
      ? undefined
      : theme;
  },
};
