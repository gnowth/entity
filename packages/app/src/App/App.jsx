import PropTypes from 'prop-types';
import React from 'react';

import { AppProvider, Context } from '../context';
import defaultHooks from './App.hooks';

function App(props) {
  const hooks = { ...defaultHooks, ...props.hooks };
  const context = React.useContext(Context);
  const { DefaultProvider, IntlProvider, ThemeProvider } = hooks.useComponents(props, context);

  const nextContext = {
    ...context,
    defaults: hooks.usePropsDefault(props, context, DefaultProvider),
    intlProviderProps: hooks.usePropsIntl(props, context, IntlProvider),
    themeProviderProps: hooks.usePropsTheme(props, context, ThemeProvider),
  };

  return (
    <AppProvider {...nextContext}>
      <IntlProvider {...nextContext.intlProviderProps}>
        <ThemeProvider {...nextContext.themeProviderProps}>
          <DefaultProvider {...nextContext.defaults}>
            { props.children }
          </DefaultProvider>
        </ThemeProvider>
      </IntlProvider>
    </AppProvider>
  );
}

App.propTypes = {
  children: PropTypes.node.isRequired,
  defaults: PropTypes.shape({}),
  hooks: PropTypes.exact({
    useComponents: PropTypes.func,
    usePropsDefault: PropTypes.func,
    usePropsIntl: PropTypes.func,
    usePropsTheme: PropTypes.func,
  }),
  intlProviderProps: PropTypes.exact({
    locale: PropTypes.string,
    messages: PropTypes.shape({}),
  }),
  themeProviderProps: PropTypes.exact({
    theme: PropTypes.shape({}),
  }),
};

App.defaultProps = {
  defaults: undefined,
  hooks: undefined,
  intlProviderProps: undefined,
  themeProviderProps: undefined,
};

export default React.memo(App);
