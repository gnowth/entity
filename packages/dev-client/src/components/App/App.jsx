import standardTheme from '@gnowth/theme-standard';
import PropTypes from 'prop-types';
import React from 'react';
import { EntityTitle } from '@entity/core';
import { AppRoot } from '@gnowth/app';
import { GlobalStyles } from '@gnowth/theme';
import { Provider } from 'react-redux';
import { IntlProvider } from '@private/react-intl';
import { ThemeProvider } from 'styled-components';

import * as theme from 'styles';

import defaults from '../../defaults';

const App = props => (
  <AppRoot
    defaults={defaults}
    entity={EntityTitle}
    intlProvider={IntlProvider}
    intlProviderProps={{ locale: 'en', messages: {} }}
    themeProvider={ThemeProvider}
    themeProviderProps={{ theme: { ...standardTheme, ...theme } }}
    {...props}
  >
    <Provider store={defaults.store}>
      { props.children }
    </Provider>

    <GlobalStyles />
  </AppRoot>
);

App.propTypes = {
  children: PropTypes.node.isRequired,
};

export default React.memo(App);
