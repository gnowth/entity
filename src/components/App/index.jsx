import PropTypes from 'prop-types';
import React from 'react';
import { AppRoot } from '@gnowth/app';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ThemeProvider } from 'styled-components';

import * as theme from 'styles';
import GlobalStyles from 'styles/global';

import defaults from '../../defaults';

const App = props => (
  <AppRoot
    defaults={defaults}
    intlProvider={IntlProvider}
    intlProviderProps={{ locale: 'en', messages: {} }}
    themeProvider={ThemeProvider}
    themeProviderProps={{ theme }}
    {...props}
  >
    <Provider store={defaults.store}>
      { props.children }
    </Provider>

    <GlobalStyles />
  </AppRoot>
);

App.propTypes = {
  children: PropTypes.node.isRequired,
};

export default App;
