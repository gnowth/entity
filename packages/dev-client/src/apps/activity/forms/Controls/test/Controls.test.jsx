import 'jest-styled-components';
import renderer from 'react-test-renderer';
import React from 'react';
import { Fields } from '@entity/core';
import { List } from 'immutable';

import App from 'components/App';
import EntityActivity from 'apps/activity/entities/Activity';

import FormControls from '..';

jest.mock('store');

it('renders correctly', () => {
  const tree = renderer.create(
    <App>
      <FormControls
        errors={List()}
        field={new Fields.EntityField({ entity: EntityActivity })}
        onChange={() => undefined}
        onSubmit={() => undefined}
        value={EntityActivity.dataToRecord({})}
      />
    </App>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
