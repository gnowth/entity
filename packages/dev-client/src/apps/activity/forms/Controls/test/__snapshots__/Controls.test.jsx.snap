// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2.c2 {
  font-size: 1rem;
}

.c7 {
  color: #ffffff;
}

.c7.c7 {
  font-size: 1rem;
}

.c3 {
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  text-transform: uppercase;
}

.c8 {
  color: #ffffff;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  text-transform: uppercase;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 0;
  border-radius: 0.25em;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  padding: 0.5em 1.5em;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  background-color: #fdfdfd;
}

.c1:focus {
  background-color: #fbfbfb;
}

.c1:disabled {
  background-color: #f8f8f8;
  cursor: auto;
}

.c1:disabled * {
  color: #e9e9e9;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 0;
  border-radius: 0.25em;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  padding: 0.5em 1.5em;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-left: 24px;
}

.c5:hover {
  background-color: #fdfdfd;
}

.c5:focus {
  background-color: #fbfbfb;
}

.c5:disabled {
  background-color: #f8f8f8;
  cursor: auto;
}

.c5:disabled * {
  color: #e9e9e9;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 0;
  border-radius: 0.25em;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  padding: 0.5em 1.5em;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #4caf50;
  min-width: 8rem;
  margin-left: 24px;
}

.c6:hover {
  background-color: #fdfdfd;
}

.c6:focus {
  background-color: #fbfbfb;
}

.c6:disabled {
  background-color: #f8f8f8;
  cursor: auto;
}

.c6:disabled * {
  color: #e9e9e9;
}

.c6:hover {
  background-color: #67bb6a;
}

.c6:focus {
  background-color: #3c9f40;
}

.c6:disabled {
  background-color: #f8f8f8;
  cursor: auto;
}

.c6:disabled * {
  color: #e9e9e9;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media print {
  .c3 {
    color: #000000;
  }
}

@media print {
  .c8 {
    color: #000000;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media print {
  .c5 {
    display: none;
  }
}

@media print {
  .c6 {
    display: none;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    content={
      Object {
        "defaultMessage": "RESET",
        "id": "packages.dev-client.src.apps.activity.forms.Controls.reset",
      }
    }
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="uiButton-Icon icon icon-name uiButton-Icon c2"
      hidden={true}
      name="name"
    />
    <span
      className="uiButton-Content c3"
      hidden={false}
    >
      RESET
    </span>
  </button>
  <div
    className="c4"
  />
  <button
    className="c5"
    content={
      Object {
        "defaultMessage": "SAVE",
        "id": "packages.dev-client.src.apps.activity.forms.Controls.save",
      }
    }
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="uiButton-Icon icon icon-name uiButton-Icon c2"
      hidden={true}
      name="name"
    />
    <span
      className="uiButton-Content c3"
      hidden={false}
    >
      SAVE
    </span>
  </button>
  <button
    className="c6"
    content={
      Object {
        "defaultMessage": "SUBMIT",
        "id": "packages.dev-client.src.apps.activity.forms.Controls.submit",
      }
    }
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="uiButton-Icon icon icon-name uiButton-Icon c7"
      hidden={true}
      name="name"
    />
    <span
      className="uiButton-Content c8"
      hidden={false}
    >
      SUBMIT
    </span>
  </button>
</div>
`;
